swift/extractor/Extractor.cpp

```diff
@@ -0,0 +1,101 @@
+#include "Extractor.h"
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <memory>
+#include <unistd.h>
+
+#include <swift/AST/SourceFile.h>
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+
+using namespace codeql;
+
+Extractor::Extractor(const Configuration& config, swift::CompilerInstance& instance)
+    : config{config}, compiler{instance} {}
+
+void Extractor::extract() {
+  if (compiler.getPrimarySourceFiles().empty()) {
+    swift::ModuleDecl* module = compiler.getMainModule();
+    if (!module->getFiles().empty() &&
+        module->getFiles().front()->getKind() == swift::FileUnitKind::Source) {
+      /// We can only call getMainSourceFile if the first file is of a Source kind
```

Comment:
```suggestion
      // We can only call getMainSourceFile if the first file is of a Source kind
```

Or is this a special convention I'm not aware of?

Reply:
Just an odd habit of mine 😃

Reply:
Yeah, so I find this slightly confusing, as in DoxyGen`///` has a special meaning. I would try to avoid this as much as possible.